src/vcpkg/commands.find.cpp

```diff
@@ -226,27 +226,31 @@ namespace vcpkg::Commands
                 print2(Color::warning, "--x-json has no effect on find artifact\n");
             }
 
-            auto metrics = LockGuardPtr<Metrics>(g_metrics);
-            metrics->track_property("command_context", "artifact");
-            if (filter.has_value())
+            auto args_hash = Hash::get_string_hash(filter.value_or_exit(VCPKG_LINE_INFO), Hash::Algorithm::Sha256);
             {
-                metrics->track_property(
-                    "command_args",
-                    Hash::get_string_hash(filter.value_or_exit(VCPKG_LINE_INFO), Hash::Algorithm::Sha256));
-            }
+                auto metrics = LockGuardPtr<Metrics>(g_metrics);
+                metrics->track_property("command_context", "artifact");
+                if (filter.has_value())
+                {
+                    metrics->track_property("command_args", args_hash);
+                }
```

Comment:
```suggestion
                if (filter.has_value())
                {
                    auto args_hash = Hash::get_string_hash(filter.value_or_exit(VCPKG_LINE_INFO), Hash::Algorithm::Sha256);
                    metrics->track_property("command_args", args_hash);
                }
```

Reply:
I don't want to call get_string_hash while holding the lock.

Reply:
then maybe:

```
//outside the lock
std::string args_hash;
if (filter.has_value())
{
    args_hash = Hash::get_string_hash(filter.value_or_exit(VCPKG_LINE_INFO), Hash::Algorithm::Sha256);
}
```

```
//inside the lock
if (!args_hash.empty())
{
    metrics->track_property("command_args", args_hash);
}
```

Reply:
Oh, I see, OK, stand by...

Reply:
OK I used .map, what do you think of this?